src/expr/src/linear.rs

```diff
@@ -1761,94 +1744,65 @@ pub mod plan {
             // Advance our lower bound to be at least the result of any lower bound
             // expressions.
             for l in self.lower_bounds.iter() {
-                // If the lower bound already exceeds a `u64`, we needn't check more.
-                if lower_bound.is_some() {
-                    match l.eval(datums, &arena) {
-                        Err(e) => {
-                            return Some(Err((e.into(), time, diff)))
-                                .into_iter()
-                                .chain(None.into_iter());
-                        }
-                        Ok(Datum::Numeric(d)) => {
-                            // If the number is negative, it cannot improve the initial
-                            // value of `Some(time)`.
-                            if !d.0.is_negative() {
-                                // An `Ok` conversion is a valid upper bound, and an `Err` error
-                                // indicates a value above `u64::MAX`. The `ok()` method does the
-                                // conversion we want to an `Option<u64>`.
-                                let v = u64::try_from(d.0).ok().map(mz_repr::Timestamp::from);
-                                // Update `lower_bound` to be the maximum with `v`, where a `None`
-                                // value is treated as larger than `Some(_)` values.
-                                lower_bound = match (lower_bound, v) {
-                                    (None, _) => None,
-                                    (_, None) => None,
-                                    (x, y) => x.max(y),
-                                };
-                            }
-                        }
-                        Ok(Datum::Null) => {
-                            null_eval = true;
-                        }
-                        x => {
-                            panic!("Non-decimal value in temporal predicate: {:?}", x);
-                        }
+                match l.eval(datums, &arena) {
+                    Err(EvalError::MzTimestampStepOverflow) => {
+                        // Since the lower bound exceeds all timestamps, it
+                        // should not appear in output.
+                        return None.into_iter().chain(None.into_iter());
+                    }
```

Comment:
I'm not sure this is correct. At least, I could imagine expressions that produce this error, but whose value isn't the max value plus one. Like, you could compare two times in an `If` block, and get this error.

Reply:
It seems not unreasonable that the output should be an error, as in the normal error case?

Reply:
Ok, removed these checks and passing through those errors. I believe this means users can't specify `<= u64::max` in their filters but...probably ok.